### PR TITLE
Build a simple bookstore management tool with React and Redux - Milestone two

### PR DESCRIPTION
**Pull request by ampaire**
In this Milestone, I worked on components and Redux store of the Build a simple bookstore management tool with React and Redux according to the [project requirements](https://github.com/microverseinc/project-redux-bookstore/blob/master/milestones/MILESTONE_2.md)

### CODE REVIEW

File: src/components/BooksList.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
```
Comment (shloch) [inline src/components/BooksList.js]: please relocate this component to `src/containers` directory ("container" components should be inside this folder)

File: src/components/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import styled from 'styled-components';
+
```
Comment (shloch) [inline src/components/BooksForm.js]: Relocate this file into `src/containers` folder too

File: src/App.js [hunk 3]
```diff
@@ -1,18 +1,12 @@
 import React from 'react';
-import './App.css';
-import styled from 'styled-components';
-
-const H4 = styled.h2`
-  font-weight: bold;
-  line-height: 52px;
-`;
+import BooksList from './components/BooksList';
+import BooksForm from './components/BooksForm';
```
Comment (shloch) [inline src/App.js]: Relocate this folder into `src/component` folder

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @ampaire , 
Screen Shot 2020-06-12 at 2 33 40 PM

- presentational components goes inside `src/components`
- container components goes inside `src/containers`

All your code logic stands...really good work. Do the minor thing below and we're good to go

You just have to relocate 3 files below.....

Happy coding

File: src/reducers/books.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const initialBooks = [
+  { id: Math.floor(Math.random() * 100), title: 'A Wrinkle in Time', category: 'Fiction' },
+  { id: Math.floor(Math.random() * 100), title: 'The Hobbit', category: 'Kids' },
+  { id: Math.floor(Math.random() * 100), title: 'Pride and Prejudice', category: 'History' },
+];
+
+const bookReducer = (state = initialBooks, action) => {
+  switch (action.type) {
+    case CREATE_BOOK:
+      return [...state, action.id];
```
Comment (shshamim63) [inline src/reducers/books.js]: Why action.id? It should be action, isn't it?

File: src/containers/BooksForm.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      <INPUT type="text" defaultValue="Add Book Title" />
+      <SELECT name="category">
+        {categories.map(category => (
+          <option key={category}>{category}</option>
+        ))}
+      </SELECT>
+      <button type="submit"> Submit </button>
+    </form>
+  );
+};
```
Comment (shshamim63) [inline src/containers/BooksForm.js]: It should have a state with title and category. Cause you will take input from the user and the user can change it at any time before submitting it. So a state is must, that will contain those values.

**Review by shshamim63 — APPROVED** [review]
## Status :+1: Approved
___
Hi @ampaire :wave: 
You have covered all the changes asked by the previous reviewer. But there are two crucial issues that the previous reviewer forgot to mention. I am sorry for that. So I am marking them as optional.
___
### :clubs: Optional Changes :clubs:
___
- The Bookform container should have a state to store the current user input status. Check out the comment.
- For create, the book should ger merged with the existing list. Not only the id. Check the comment
- The linter is not working properly. Please fix it by following the [link](https://github.com/microverseinc/linters-config/tree/master/react-redux).
____
##### This :+1: is from me.
____
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by ampaire** [comment]
Thank you for the insight @shshamim63

**Pull request merged**